### PR TITLE
Report triggered schema tests status separately.

### PR DESCRIPTION
**Pull request by danielroseman**
For downstream builds triggered by govuk-content-schemas, send the
pending/success/failed notification to the triggering commit on the
schemas repo. This will show the status individually in the Github UI
and permit re-running of individual tests where necessary.

(This will also need a change to the schemas test itself, to pass the commit sha and not wait for the downstream results.)

https://trello.com/c/5psKmk85

### CODE REVIEW

**Comment by danielroseman** [comment]
@tijmenb @suzannehamilton what do you think?

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 1]
```diff
@@ -393,7 +393,11 @@ def buildProject(sassLint = true) {
         [$class: 'StringParameterDefinition',
           name: 'SCHEMA_BRANCH',
```
Comment (suzannehamilton) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: I wonder if we could rid of this parameter and use `SCHEMA_COMMIT` everywhere instead?

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 2]
```diff
@@ -526,4 +540,25 @@ def hasDatabase() {
...
+ * Manually set build status in Github.
+ *
+ * Useful for downstream builds that want to report on the upstream PR.
+ *
+ * @param repoName Name of the repo being built
+ * @param commit SHA of the triggering commit on govuk-content-schemas
+ * @param message The message to report
+ * @param state The build state: one of PENDING, SUCCESS, FAILED
+ */
+def setBuildStatus(repoName, commit, message, state) {
```
Comment (suzannehamilton) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: Just wrapping my head around this - this reports the build status against the content schemas, so a build error will show up on the correct govuk-content-schemas PR?

Comment (danielroseman) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: Yes, that's the intention - see https://github.com/alphagov/govuk-content-schemas/pull/572 for an example.

**Comment by suzannehamilton** [comment]
This looks great! 👏

**Comment by tijmenb** [comment]
This looks like a great solution! Really happy with this.

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy (hunk 1 above)
Comment (danielroseman) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: Potentially yes, but it'll be messy if we try to do that before all the dependent apps are moved over to use buildProject so probably continue with both params until then.

**Pull request merged**